### DISCUSSION TITLE
TDS: IEEE14 bus system, Use Case: 1.1 demo_DGPRCT1.ipynb

### DISCUSSION THREAD

**Discussion by JunejaAmmy** [CURENT/andes#255] [Q&A] [answered]
Hi Dr. Cui,

I followed the **1.1 demo_DGPRCT1.ipynb** use case. After running the TDS, I'm plotting the graph using following command:

`ss.TDS.plt.plot(ss.TGOV1.LAG_y)`

It plotted the graph as shown below:

![image](https://user-images.githubusercontent.com/97614806/150842417-29e869d3-8850-4a38-b3a1-097c470be2ba.png)

I checked the data in variable **ss.TGOV1.LAG_y**, it contains following data: 

![image](https://user-images.githubusercontent.com/97614806/150842742-9d3d074b-3a18-4378-891e-185ef5ea8a13.png)

The data contained in **array v** seems to be the starting point of the lines. Please correct me If I'm wrong.

My query is:
How the data in the variable **ss.TGOV1.LAG_y** is drawing the complete lines? Please guide me.

**Comment by cuihantao** [comment, marked as answer]
`ss.TGOV1.LAT_y.v` contains the data for the very last time stamp. 

The full data can be retrieved at `ss.dae.ts.x[:, ss.TGOV1.LAG_y.a]`, and the corresponding timestamps are in `ss.dae.ts.t`.

Also see https://github.com/cuihantao/andes/blob/master/examples/2.%20inspect_data.ipynb, although it needs improving by adding examples.

**Comment by JunejaAmmy** [comment]
Thank you for your response,

I checked the data at `ss.dae.ts.x[ss.TGOV1.LAG_y.a]`. It contains all exponential values as shown below:

![image](https://user-images.githubusercontent.com/97614806/150857981-23c929b2-be37-4aae-b762-015b2a4f98bf.png)

It does not seem to have any value to draw the lines of the above mentioned plot.

**Reply by cuihantao** [reply]
Sorry, it should be `ss.dae.ts.x[:, ss.TGOV1.LAG_y.a]`. Notice that colon and comma.

**Reply by JunejaAmmy** [reply]
Thanks, now I'm able to see the relative points.

![image](https://user-images.githubusercontent.com/97614806/150859617-6dc65b62-8ef4-4989-a853-1cd67e196195.png)

Please have a look at the blue-highlighted dots. It doesn't seems to be showing the whole data to draw the complete lines.

**Reply by cuihantao** [reply]
This is how NumPy prints the data. Data is in there but just not printed. Feel free to search for NumPy tutorials that might help.

**Reply by JunejaAmmy** [reply]
Thanks a million,

I got the solution, now I'm able to print the whole data.

![image](https://user-images.githubusercontent.com/97614806/150861169-0b66a5c3-8d30-4669-8a07-e268052403ff.png)

**Reply by cuihantao** [reply]
Great. You are welcome. You can mark the discussion as resolved.

**Reply by JunejaAmmy** [reply]
yeah sure.
